src/Elliptic/Systems/ScalarGaussBonnet/Equations.cpp

```diff
@@ -0,0 +1,147 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/ScalarGaussBonnet/Equations.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/EagerMath/RaiseOrLowerIndex.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace sgb {
+
+void curved_fluxes(const gsl::not_null<tnsr::I<DataVector, 3>*> flux_for_field,
+                   const tnsr::II<DataVector, 3>& inv_conformal_metric,
+                   const tnsr::I<DataVector, 3>& shift,
+                   const Scalar<DataVector>& lapse,
+                   const Scalar<DataVector>& conformal_factor,
+                   const tnsr::i<DataVector, 3>& field_gradient) {
+  raise_or_lower_index(flux_for_field, field_gradient, inv_conformal_metric);
+
+  for (size_t i = 0; i < 3; i++) {
+    flux_for_field->get(i) /= pow<4>(get(conformal_factor));
+    flux_for_field->get(i) -= get(dot_product(shift, field_gradient)) *
```

Comment:
Move the dot product divided by lapse out of the loop, so it isn't repeated three times. Or add a note that it's a possible performance optimization.

Reply:
Ok it's not obvious that moving psi^4 out of the loop is faster because it does an extra memory allocation now. The dot product does that anyway, so moving it out of the loop is clearly an optimization. I'd say don't worry about it until you do a profiling + performance optimization pass over the solver.